src/GraphODataTemplateWriter/CodeHelpers/Java/TypeHelperJava.cs

```diff
@@ -753,6 +752,29 @@ public static string CreatePackageDefForEnum(this CustomT4Host host)
             sb.Append("\n");
             return sb.ToString();
         }
+        public static string GetPackagePrefix(this OdcmObject obj)
+        {
+            if (obj is OdcmEnum)
+            {
+                return "models.generated";
+            }
+            else if (obj is OdcmType)
+            {
+                return GetPrefixForModels();
+            }
+            else if (obj is OdcmParameter)
+            {
+                return GetPackagePrefix((obj as OdcmParameter).Type);
+            }
+            else if (obj is OdcmProperty)
+            {
+                return GetPackagePrefix((obj as OdcmProperty).Type);
+            }
+            else
+            {
+                throw new ArgumentException("obj");
+            }
```

Comment:
nit: `is` -> `as` can be bypassed by:
```suggestion
            switch(obj)
            {
                case OdcmEnum e:
                    return "";
                case OdcmType t:
                    return GetPrefixForModels();
                case OdcmParameter p:
                    return GetPackagePrefix(p.Type);
                case OdcmProperty p:
                    return GetPackagePrefix(p.Type);
                default:
                    throw new ArgumentException(nameof(obj));
            }
```